src/basic/string-util.c

```diff
@@ -145,57 +145,32 @@ char *strnappend(const char *s, const char *suffix, size_t b) {
 
 char *strjoin_real(const char *x, ...) {
         va_list ap;
-        size_t l;
+        size_t l = 0;
         char *r, *p;
 
         va_start(ap, x);
+        for (const char *t = x; t; t = va_arg(ap, const char *)) {
+                size_t n;
 
-        if (x) {
-                l = strlen(x);
-
-                for (;;) {
-                        const char *t;
-                        size_t n;
-
-                        t = va_arg(ap, const char *);
-                        if (!t)
-                                break;
-
-                        n = strlen(t);
-                        if (n > ((size_t) -1) - l) {
-                                va_end(ap);
-                                return NULL;
-                        }
-
-                        l += n;
+                n = strlen(t);
+                if (n > ((size_t) -1) - l) {
```

Comment:
when you touch this maybe also change (size_t) -1 → SIZE_MAX. It's more readable I think, and doesn't imply 2's-complement integer logic.

Reply:
Agree on the "more readable" part, and I'll update it. Just noting that (size_t)-1 outnumbers SIZE_MAX about 4:1 currently, and the 2's complement is somewhat academic since no compiler capable of building a linux kernel supports any other integer representations.

Reply:
Actually, to beat a dead horse, it has nothing to do with 2's complement or not. C99, 6.3.1.3.2, "Otherwise, if the new type is unsigned, the value is converted by repeatedly adding or subtracting one more than the maximum value that can be represented in the new type until the value is in the range of the new type.", so for any unsigned type T [except _Bool], `(T)-1` gets converted by adding typemax(T)+1 once, arriving at typemax(T).

Reply:
Wouldn't it be easier to say that the value is taken `mod typemax(T)+1` ;) ?